test/test_joint.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import absolute_import, division, print_function
```

Comment:
There aren't many tests here, but since this PR moves much of the Gaussian functionality into `Joint`, we can rely on the `Gaussian` unit tests for many of the tests.